사이드바 디자인 변경 및 트랜지션을 추가한다

voidmatcha opened the pull request:
> resolve #24

## Detail & Solution
은비님이 MS Edge 브라우저의 사이드바 트랜지션 느낌으로 디자인을 변경하신 부분을 반영하였습니다.

https://user-images.githubusercontent.com/16266103/111889965-81722b00-8a28-11eb-974d-6ed19274e812.mp4

<div>
<img width="90" alt="Screen Shot 2021-03-21 at 9 32 10 AM" src="https://user-images.githubusercontent.com/16266103/111889942-52f45000-8a28-11eb-89f4-c19a5620c7c5.png">
<img width="72" alt="Screen Shot 2021-03-21 at 9 31 31 AM" src="https://user-images.githubusercontent.com/16266103/111889935-3bb56280-8a28-11eb-9c3d-1ebfd37047d3.png">
</div>

## What I did
- styled-components 자체에서 vender prefix를 자동으로 추가해주는데 이를 임의로 적용한 스타일이 있어 제거하였습니다.
- 기존에 js로 관리되던 hover 상태 및 트랜지션을 CSS 기반 트랜지션으로 변경하였습니다.
- 스타일 리펙토링 및 사이드바 버튼 Text를 제거하였습니다.
- #13 에서 JWT 기반으로 변경되어 새로고침을 해도 해당 페이지에 머물 수 있도록 리팩토링 되었습니다. 하지만 새로고침시 SideBar가 버튼이 선택된 위치를 기억하지 못하는 이슈가 있어 url path를 가져와 url에 따라 올바른 버튼이 매핑되어 선택되게끔 수정하였습니다. https://github.com/witherview/witherview_frontend/pull/32/commits/2a8cf2a213cb504a64c73838277a0793aabcdd2b
- 새로고침시 SideBar의 올바른 위치의 버튼이 선택될 수 있도록 하는 로직을 위해 path의 구조를 `/` 를 기준으로 통일하도록 변경하였습니다. https://github.com/witherview/witherview_frontend/pull/32/commits/d380875dda3e00ff4a00ebcdba612baf5948fb50



### Before
[Storybook: SideBar - Before](https://dididy.github.io/witherview_frontend/?path=/story/organisms-side-bar--side-bar)

https://user-images.githubusercontent.com/16266103/113977783-0f01a780-987e-11eb-82de-cb02027e636d.mov

### After
[Storybook: SideBar - After](https://dididy.github.io/witherview_frontend/refactor/sidebar/?path=/story/organisms-side-bar--side-bar)

https://user-images.githubusercontent.com/16266103/113978055-6dc72100-987e-11eb-8112-944c68ff2959.mov

## What I need to do

voidmatcha committed:
refactor: 💡 사이드바 디자인 변경 및 트랜지션을 추가한다 (#24)

- styled-components 자체에서 vender prefix를 추가해지므로 관련된 css 제거
- js로 관리되던 hover 상태 및 트랜지션을 제거
  - CSS 기반 트랜지션으로 변경
- 사이드바 버튼 Text 제거
- 스타일 리팩토링
✅ Closes: #24

wooooooood reviewed (approved):
영상까지 첨부해주시다니 감사합니다!👍

voidmatcha committed:
refactor: 💡 새로고침시 url path로 SideBar 버튼 선택을 매핑하도록 한다

- useLocation으로 url path를 가져오도록 함
- 상태명을 click에서 path로 변경
- path에 맞춰 선택 및 아이콘 렌더링을 결정하도록 props 변경
- 새로고침시 url path를 기반으로 버튼 선택을 매핑하도록 함

voidmatcha committed:
refactor: 💡 SideBar 버튼 기준으로 react-router-dom path 를 수정한다

- Route에 exact 프로퍼티 누락된 부분 추가
- SideBar 기준으로 페이지 넘어갈 때 / 로 구분하도록 path 수정
- 네이밍 변경 (video => replay)

voidmatcha commented on src/components/organisms/Sidebar/SideBar.js:
```diff
@@ -56,21 +56,25 @@ const WrapBottomButton = styled.div`
+    }
+    history.push('/self');
+    return null;
+  }, []);
```
#13 에서 JWT 기반으로 새로고침을 해도 해당 url에 머물 수 있게끔 리팩토링 하였는데 사이드바가 선택된 버튼의 위치를 기억하지 못하는 이슈가 있어서 url을 가져와 매핑하여 선택되어질 수 있도록 수정하였습니다.

voidmatcha commented on src/App.js:
```diff
@@ -82,25 +82,26 @@ export default function App() {
             />
-            <R.AuthRoute path="/question/:id" component={QuestionPage} />
+            <R.AuthRoute exact path="/self/question/:id" component={QuestionPage} />
             <R.AuthRoute
+              exact
               path="/self/setting/:id"
               component={SelfTrainSettingPage}
             />
-            <R.AuthRoute path="/self-train/:id" component={SelfTrainPage} />
+            <R.AuthRoute exact path="/self/train/:id" component={SelfTrainPage} />
             <R.AuthRoute
               exact
-              path="/self-checklist/:roomId"
+              path="/self/checklist/:roomId"
               component={SelfStudyChecklist}
             />
-            <R.AuthRoute exact path="/myvideo" component={MyVideoPage} />
-            <R.AuthRoute exact path="/video/:id" component={VideoPage} />
-            <R.AuthRoute path="/peer-study" component={PeerStudyMainPage} />
-            <R.AuthRoute path="/peer/:id" component={R.PeerStudyRoute} />
-            <R.AuthRoute path="/mypage" component={MyPage} />
+            <R.AuthRoute exact path="/replay" component={MyVideoPage} />
+            <R.AuthRoute exact path="/replay/:id" component={VideoPage} />
+            <R.AuthRoute exact path="/peer-study" component={PeerStudyMainPage} />
+            <R.AuthRoute exact path="/peer-study/:id" component={R.PeerStudyRoute} />
+            <R.AuthRoute exact path="/mypage" component={MyPage} />
```
누락했던 exact 프로퍼티를 추가하였습니다.

https://github.com/witherview/witherview_frontend/blob/d380875dda3e00ff4a00ebcdba612baf5948fb50/src/components/organisms/Sidebar/SideBar.js#L66-L73

새로고침시 SideBar의 올바른 위치의 버튼이 선택될 수 있도록 하는 로직을 위해 path의 구조를 `/` 를 기준으로 통일하도록 변경하였습니다.

voidmatcha reviewed (commented):
#13 에서 발견한 버그를 해결하고 react-router-dom의 path를 통일성 있게 리팩토링 하였습니다.

Sung-jin commented on src/components/organisms/Sidebar/SidebarButton/SideBarButton.js:
```diff
@@ -21,22 +21,10 @@ const Wrapper = styled.button`
-  color: ${({ clicked }) => (clicked ? '#0c0c59' : 'white')};
-`;
-
-export default function SideBarButton({
```
위 형태가 클래스 컴포넌트에서 

```
constructor(props) {
        super(props);
...
}

...

this.props....
```

형태와 같은건가요?
class component 대용으로 hooks 로 다 해결이 가능하다고 들었는데, 해당 내용이 hooks 인가요?

hooks 내용을 제대로 본건 아니지만.. 위 형태는 아닌거 같은데 다른 문법인가요??
키워드만 알려주신다면 찾아보겠습니다!

Sung-jin replied:
SideBarButton.propTypes 에 값을 정의하는거 보면.. styled component 같은 라이브러리 안건가요?

voidmatcha replied:
네네 함수형컴포넌트로 사용하면 클래스 컴포넌트로 사용할 때 constructor로 받았던 props를 destructuring으로 손쉽게 받을 수 있습니다!

https://medium.com/@lcriswell/destructuring-props-in-react-b1c295005ce0

propTypes는 리엑트에서 제공해주는 타입체커입니다. 별도의 라이브러리는 아니에요!

https://reactjs-kr.firebaseapp.com/docs/typechecking-with-proptypes.html

Sung-jin reviewed (commented):
고생하셨습니다 👍

Sung-jin reviewed (approved):
approve 를 한다는걸.. comment 로 했었나 보네요. 👍
